assets/js/components/flows/FlowsWorkspace.jsx

```diff
@@ -1,151 +1,212 @@
+    let data = { label: name, id };
+    if (type === "channelNode") {
       data = Object.assign({}, data, {
-        type_name: event.dataTransfer.getData('node/channel_type_name'),
-        type: event.dataTransfer.getData('node/channel_type')
-      })
+        type_name: event.dataTransfer.getData("node/channel_type_name"),
+        type: event.dataTransfer.getData("node/channel_type"),
+      });
     }
 
-    if (type === 'functionNode') {
+    if (type === "functionNode") {
       data = Object.assign({}, data, {
-        format: event.dataTransfer.getData('node/function_format'),
-      })
+        format: event.dataTransfer.getData("node/function_format"),
+      });
     }
 
-    if (type === 'labelNode') {
+    if (type === "labelNode") {
       data = Object.assign({}, data, {
-        deviceCount: event.dataTransfer.getData('node/label_device_count'),
-      })
+        deviceCount: event.dataTransfer.getData("node/label_device_count"),
+      });
     }
 
-    const newNode = { id, type, position, data }
+    const newNode = { id, type, position, data };
 
-    if (type === 'functionNode' && elementsMap[id]) {
-      let i = 1
+    if (type === "functionNode" && elementsMap[id]) {
+      let i = 1;
       while (elementsMap[id + "_copy" + i]) {
-        i ++
+        i++;
       }
-      newNode.id = id + "_copy" + i
+      newNode.id = id + "_copy" + i;
     }
 
-    setElements(elsMap => Object.assign({}, elsMap, { [newNode.id]: newNode }))
-    setChangesState(true)
-  }
+    setElements((elsMap) =>
+      Object.assign({}, elsMap, { [newNode.id]: newNode })
+    );
+    setChangesState(true);
+  };
 
   const onAdrUpdate = (id, adrAllowed) => {
-    const newNodeData = Object.assign({}, elementsMap[id].data, { adrAllowed })
-    const newNode = Object.assign({}, elementsMap[id], { data: newNodeData })
-    setElements(elsMap => Object.assign({}, elsMap, { [newNode.id]: newNode }))
-  }
+    const newNodeData = Object.assign({}, elementsMap[id].data, { adrAllowed });
+    const newNode = Object.assign({}, elementsMap[id], { data: newNodeData });
+    setElements((elsMap) =>
+      Object.assign({}, elsMap, { [newNode.id]: newNode })
+    );
+  };
 
   const onMultiBuyUpdate = (id, multi_buy_id) => {
-    const newNodeData = Object.assign({}, elementsMap[id].data, { multi_buy_id })
-    const newNode = Object.assign({}, elementsMap[id], { data: newNodeData })
-    setElements(elsMap => Object.assign({}, elsMap, { [newNode.id]: newNode }))
-  }
+    const newNodeData = Object.assign({}, elementsMap[id].data, {
+      multi_buy_id,
+    });
+    const newNode = Object.assign({}, elementsMap[id], { data: newNodeData });
+    setElements((elsMap) =>
+      Object.assign({}, elsMap, { [newNode.id]: newNode })
+    );
+  };
+
+  const onAlertUpdate = (id, type, hasAlerts) => {
+    if (type === "function") {
+      let functionNodes = {};
+      for (const [key, _value] of Object.entries(elementsMap)) {
```

Comment:
so clean

Reply:
i'm proud of this one, took like 5 rounds of refactor lols